### DISCUSSION TITLE
Extraction naming consistency

### DISCUSSION THREAD

**Discussion by AntonioSun** [smicyk/groovy-jmeter#93] [Ideas]
I know this is kind of nitty-picky, but hear me out:

https://github.com/smicyk/groovy-jmeter/blob/07b89a2c4ff0a353d5033edca610031f7085a6b6/src/test/groovy/net/simonix/dsl/jmeter/factory/extractor/ExtractorFactorySpec.groovy#L30-L32

I.e., both `extract_regex` and `extract_jmes` use `variable` and `expression`.
Yet, `extract_json` uses `variables` and `expressions` (both plural) instead.

I just feel that the plurals here don't make much sense, and the naming is not consistent. 
However, I do understand that you might have many legacy code done this way.

**Comment by smicyk** [comment]
Hi, the reason for that is the extract_json can have multiple expression and variables e.g. 

```groovy
extract_json(variables: ['var_variable1', 'var_variable2'], expressions: ['$..author', '$..book'])
```

but you if have only one variable and expression then you can use string only without []

```groovy
extract_json(variables: 'var_variable', expressions: '$..author')
```
I would agree that we can split it into two so you can use variable/expression if you have only one and variables/expressions if you have multiple. But yeah I don't have strong opinion in that.

**Reply by AntonioSun** [reply]
Oh, in that case, let it be then. Thanks for the explanation.
